File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  attr_reader :square_of_sum, :sum_of_squares, :difference

  def initialize(n)
    @square_of_sum = (1..n).reduce(:+) ** 2
    @sum_of_squares = (1..n).map { |x| x ** 2 }.reduce(:+)
    @difference = @square_of_sum - @sum_of_squares
  end
end

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
    constructor(n) {
        this.values = [];

        for (let i = 1; i <= n; ++i)
            this.values.push(i);
    }

    get sumOfSquares() {
        return this.values.reduce((acc, val) => acc + val ** 2);
    }

    get squareOfSum() {
        return this.values.reduce((acc, val) => acc + val) ** 2;
    }

    get difference() {
        return this.squareOfSum - this.sumOfSquares;
    }
}
